### ISSUE TITLE
Link anchors are not ignored

### ISSUE THREAD

**Issue by fidian** [emmercm/metalsmith-link-checker#50]
In HTML,  I link to different headings.

```html
<a name="TableOfContents"><h1>Table of Contents</h1></a>

<p>Lots of content</p>

<a href="#TableOfContents">Back to the Table of Contents</a>
```

When I use metalsmith-link-checker, it fails saying messages like this:

```
user-guide.html:
  #TableOfContents (not found)
```

**Comment by emmercm** [comment]
Hi @fidian , are the header links purely within-page like the example you gave, or cross-page?

Anchors should be stripped off of links, I'm not validating them right now: https://github.com/emmercm/metalsmith-link-checker/blob/59a3af97371599c43d9ddd787f390f9e1feed6b0/lib/index.js#L244-L251

Does there happen to be extra, stray characters in your link?

**Comment by fidian** [comment]
I'm using ones like my examples. They link to the same page.

As for stray characters, I didn't see any. I'm using [`metalsmith-bookify-html`](https://github.com/fidian/metalsmith-bookify-html) and the links are in the generated single-page document.

**Comment by emmercm** [comment]
Copying and pasting your example into a new unit test I don't see any issues: https://github.com/emmercm/metalsmith-link-checker/pull/51/files#diff-7e6059ce8f0a008d7d1c2235a5a0fac5f913723d493ff327ffedf732188eb277R13-R16

Are you on the latest version?

**Comment by fidian** [comment]
I have created [a branch in metalsmith-site-example](https://github.com/fidian/metalsmith-site-example/tree/metalsmith-link-checker-50) that should show off what I have configured.

**Comment by fidian** [comment]
Sorry, I neglected to give instructions. There's the obvious `git clone` and `npm install`. After that, `npm run fastbuild` runs a build without the link checker and `npm run build` does one with the link checker. That's the only difference between the two .

**Comment by emmercm** [comment]
I think your project might need an `npm i --save metalsmith`, as a heads-up.

I ran the branch and my output was:

```shell
$ npm run build   

> @fidian/metalsmith-site-example@ build /private/var/folders/9n/lb2q4q1d5xq89tst4vh_7_br0000gp/T/tmp.kTkr3B6n
> node metalsmith.js

Build 1 started
Broken links found:

book.html:
  #events/2021-01-01-new-year-day/index.html (not found)
  #events/2021-02-02-groudhog-day/index.html (not found)
  #events/2021-02-14-valentines-day/index.html (not found)
  #events/index.html (not found)
  ../../css/reset.css (not found)
```

It's because of the `/` in the fragment identifier. I'll have a fix out today.

**Comment by emmercm** [comment]
Fixed with #51, new version is published: https://www.npmjs.com/package/metalsmith-link-checker/v/0.5.0

**Comment by fidian** [comment]
Thanks!
